### PR TITLE
Add inline type annotations

### PR DESCRIPTION
**Pull request by Pankrat**
Since this package is only compatible with Python 3, add type
annotations for all function definitions and check this via mypy.

A few changes were necessary to fix errors identified by mypy:

* Drop the unnecessary sentinel object `settings.DEFAULT` and use the
default broker string directly in function definitions.
* Check if the channel/connection is open in various transport methods.
* Drop unused `pending` and `messages` attributes in `Transport`.

### CODE REVIEW

File: domain_event_broker/events.py [hunk 1]
```diff
@@ -14,13 +15,13 @@ def to_timestamp(dt, epoch=datetime(1970, 1, 1)):
...
 
     def __init__(self,
-                 routing_key=u"",
-                 data={},
-                 domain_object_id=None,
-                 uuid_string=None,  # only set if recreated from json repr
-                 timestamp=None,  # only set if recreated from json repr
-                 retries=0,
-                 **kwargs):
+                 routing_key: str = u"",
```
Comment (stedi67) [inline domain_event_broker/events.py]: We don't need the `u""` any longer.

File: domain_event_broker/events.py [hunk 2]
```diff
@@ -52,24 +53,24 @@ def __init__(self,
...
 
-    def __repr__(self):
-        return u"DomainEvent('{0.routing_key}', '{0.data}', (domain_obj_id: {0.domain_object_id}, uuid: {0.uuid_string}))".format(self) # noqa
+    def __repr__(self) -> str:
+        return "DomainEvent('{0.routing_key}', '{0.data}', (domain_obj_id: {0.domain_object_id}, uuid: {0.uuid_string}))".format(self) # noqa
 
     __str__ = __repr__
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
```
Comment (stedi67) [inline domain_event_broker/events.py]: Shouldn't we type `other` as `DomainEvent` here?

Comment (Pankrat) [reply domain_event_broker/events.py]: I had that originally and this is what mypy had to say about it:

```
domain_event_broker/events.py:73: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"
domain_event_broker/events.py:73: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
domain_event_broker/events.py:73: note:     def __eq__(self, other: object) -> bool:
domain_event_broker/events.py:73: note:         if not isinstance(other, DomainEvent):
domain_event_broker/events.py:73: note:             return NotImplemented
domain_event_broker/events.py:73: note:         return <logic to compare two DomainEvent instances>
Found 1 error in 1 file (checked 24 source files)
```

I didn't see any good reasons to do it differently.

Comment (stedi67) [reply domain_event_broker/events.py]: Well, who'd want to argue with mypy...

**Pull request merged**